### DISCUSSION TITLE
Standardize platform specific code

### DISCUSSION THREAD

**Discussion by karol-bisztyga** [software-mansion/react-native-reanimated#1523] [Ideas]
For now, we have some platform-specific code in our codebase. The problem is it occurs wherever it's needed in different places. Some examples:

Checks in files:
https://github.com/software-mansion/react-native-reanimated/blob/master/src/reanimated2/core.js#L97
https://github.com/software-mansion/react-native-reanimated/blob/master/src/reanimated2/Colors.js#L578
https://github.com/software-mansion/react-native-reanimated/blob/master/src/reanimated2/Hooks.js#L45

Platform specific files:
https://github.com/software-mansion/react-native-reanimated/tree/master/src/reanimated2/platform-specific
https://github.com/software-mansion/react-native-reanimated/tree/master/src/reanimated2 - NativeReanimated

I think it would be good to put all of those in one file/directory plus explicitly and clearly distinguish certain versions of code for every platform.

If we want to support more platforms in the future it will be also easier to implement.

**Comment by ArMANIAK** [comment]
Hello.
During small hackathon Wix Goodness Squad I've worked with this issue. You may check it https://github.com/ArMANIAK/react-native-reanimated/tree/feat-platform-specific-code-separation
If it's ok feel free to merge and use (not sure whether I should open PR). Unfortunatelly, I have doubts I can be helpfull any more

**Reply by karol-bisztyga** [reply]
Hey! Thanks for the efforts! Please, open a pull request in our repo so we can clearly see changes and discuss them there, thanks!

**Reply by ArMANIAK** [reply]
Done. You may see it here: https://github.com/software-mansion/react-native-reanimated/pull/1558
